src/cljam/io/csi.clj

```diff
@@ -0,0 +1,76 @@
+(ns cljam.io.csi
+  "Reader of a CSI format file."
+  (:require [cljam.io.util.bgzf :as bgzf]
+            [cljam.io.util.lsb :as lsb]
+            [cljam.io.util.bin :as util-bin])
+  (:import java.util.Arrays
+           [java.io DataInputStream IOException]
+           [cljam.io.util.chunk Chunk]))
+
+(deftype CSI [n-ref min-shift depth bidx loffset]
+  util-bin/IBinaryIndex
+  (get-chunks [_ ref-idx bins]
+    (into [] (mapcat (get bidx ref-idx) bins)))
+  (get-min-offset [_ ref-idx beg bins]
```

Comment:
I'm sorry I forgot that CSI doesn't always store l-offsets for all bins like BAI.
So in this comment https://github.com/chrovis/cljam/pull/182#discussion_r348276378, when bin `12` doesn't exist in CSI, we have to check the left bin `11` and then the parent bin `5` ... recursively until we find an existing bin.
Note that the bin `11` is not contained in the result of `reg->bins` in this case.
The current implementation will return the l-offset of the bin `13` if `12` doesn't exist, which results in unintentionally filtering necessary chunks.